File: src/mvmctl/models/binary.py
"""Binary data models."""

from __future__ import annotations

from dataclasses import dataclass
from typing import TYPE_CHECKING

from mvmctl.utils.common import CommonUtils

if TYPE_CHECKING:
    from mvmctl.models.vm import VMInstanceItem


@dataclass
class BinaryItem:
    """
    Binary record — maps to binaries table.

    The ``path`` field stores the absolute filesystem path to the binary
    (e.g. ``"/home/user/.cache/mvmctl/bin/firecracker-v1.15.1"``).
    """

    id: str
    name: str
    version: str
    full_version: str
    ci_version: str | None
    path: str
    is_default: bool
    is_present: bool
    created_at: str
    updated_at: str
    deleted_at: str | None = None

    vms: list[VMInstanceItem] | None = None

    def __post_init__(self) -> None:
        """Coerce bool fields loaded from SQLite."""
        CommonUtils.coerce_bool_fields(self, {"is_default", "is_present"})

File: internal/infra/model/binary.go
package model

// ── BinaryItem ──

// BinaryItem corresponds to Python's BinaryItem dataclass.
type BinaryItem struct {
	ID          string  `json:"id" db:"id"`
	Name        string  `json:"name" db:"name"`
	Version     string  `json:"version" db:"version"`
	FullVersion string  `json:"full_version" db:"full_version"`
	CIVersion   *string `json:"ci_version,omitempty" db:"ci_version"`
	Path        string  `json:"path" db:"path"`
	IsDefault   bool    `json:"is_default" db:"is_default"`
	IsPresent   bool    `json:"is_present" db:"is_present"`
	CreatedAt   string  `json:"created_at" db:"created_at"`
	UpdatedAt   string  `json:"updated_at" db:"updated_at"`
	DeletedAt   *string `json:"deleted_at,omitempty" db:"deleted_at"`

	// Resolved relations
	VMs []*VM `json:"vms,omitempty"`
}
